allensdk/brain_observatory/filtered_running_speed/__main__.py

```diff
@@ -0,0 +1,954 @@
+
+    if num_raw_timestamps != len(signal):
+        raise ValueError(
+            f"found {num_raw_timestamps} rising edges on the vsync line, "
+            f"but found {len(signal)} samples for  {label}"
+        )
+
+    return signal
+
+
+def extract_dx_info(
+    frame_times,
+    start_index,
+    end_index,
+    pkl_path
+):
+    """
+    Extract all of the running speed data
+
+    Parameters
+    ----------
+    frame_times: numpy.ndarray
+        list of the vsync times
+    start_index: int
+        Index to the first frame of the stimulus
+    end_index: int
+       Index to the last frame of the stimulus
+    pkl_path: string
+        Path to the stimulus pickle file
+
+    Returns
+    -------
+    list[pd.DataFrame, pd.DataFrame]
+        the velocity data and the raw data
+
+    Notes
+    -------
+            velocity pd.DataFrame:
+                columns:
+                    "velocity": computed running speed
+                    "net_rotation": dx in radians
+                    "frame_indexes": frame indexes into
+                        the full vsync times list
+
+            raw data pd.DataFrame:
+                Dataframe with an index of timestamps and the following
+                columns:
+                    "vsig": voltage signal from the encoder
+                    "vin": the theoretical maximum voltage that the encoder
+                        will reach prior to "wrapping". This should
+                        theoretically be 5V (after crossing 5V goes to 0V, or
+                        vice versa). In practice the encoder does not always
+                        reach this value before wrapping, which can cause
+                        transient spikes in speed at the voltage "wraps".
+                    "frame_time": list of the vsync times
+                    "dx": angular change, computed during data collection
+                The raw data are provided so that the user may compute their
+                own speed from source, if desired.
+
+    """
+
+    stim_file = pd.read_pickle(pkl_path)
+
+    frame_times = frame_times[start_index:end_index]
+
+    # occasionally an extra set of frame times are acquired after the rest of
+    # the signals. We detect and remove these
+    frame_times = sync_utilities.trim_discontiguous_times(frame_times)
+    num_raw_timestamps = len(frame_times)
+
+    dx_deg = running_from_stim_file(stim_file, "dx", num_raw_timestamps)
+    vsig = running_from_stim_file(stim_file, "vsig", num_raw_timestamps)
+    vin = running_from_stim_file(stim_file, "vin", num_raw_timestamps)
+
+    dx_deg = match_timestamps(num_raw_timestamps, dx_deg, 'dx')
+    vsig = match_timestamps(num_raw_timestamps, vsig, 'vsig')
+    vin = match_timestamps(num_raw_timestamps, vin, 'vin')
+
+    velocities = extract_running_speeds(
+        dx_deg,
+        frame_times,
+        vsig,
+        vin,
+        start_index,
+        end_index,
+        USE_LOWPASS_FILTER,
```

Comment:
Should `USE_LOWPASS_FILTER` and `DEFAULT_ZSCORE_THRESHOLD` be exposed in the call signature of`extract_dx_info`?

Reply:
This has been refactored.